src/sparseml/optim/helpers.py

```diff
@@ -247,7 +250,10 @@ def evaluate_recipe_yaml_str_equations(recipe_yaml_str: str) -> str:
                 val, variables, non_val_variables
             )
 
-    return rewrite_recipe_yaml_string_with_classes(container)
+    # I find it really helpful to use the container returned by this function.
+    # I'd love to either make this return only the container
+    # or the (func(container), container)
+    return rewrite_recipe_yaml_string_with_classes(container), container
```

Comment:
Better `return container` and pushing out `rewrite_recipe_yaml_string_with_classes`?

Reply:
why would we prefer that here? this is a top level function that takes in a valid SparseML recipe, so it should also return one. the container here will not have the proper `Modifier` classes loaded

Reply:
also if the only purpose of returning container here is so we can get a loaded dict of the yaml string (and don't actually care about the variables, just use `load_recipe_yaml_str_no_classes`